Arf; looks like I need to update a test;

```
=== RUN   TestDockerSuite/TestBuildAddFileNotFound
    --- FAIL: TestDockerSuite/TestBuildAddFileNotFound (0.16s)
        docker_cli_build_test.go:2201: assertion failed: 
            Command:  /usr/local/cli/docker build -t testbuildaddnotfound .
            ExitCode: 1
            Error:    exit status 1
            Stdout:   Sending build context to Docker daemon  3.072kB

            Step 1/2 : FROM scratch
             ---> 
            Step 2/2 : ADD foo /usr/local/bar
            
            Stderr:   ADD failed: file not found in build context or excluded by .dockerignore: stat foo: file does not exist
            
            
            Failures:
            Expected stderr to contain "foo: no such file or directory"
```

--- a/builder/dockerfile/copy.go
+++ b/builder/dockerfile/copy.go
@@ -242,6 +242,8 @@ func (o *copier) calcCopyInfo(origPath string, allowWildcards bool) ([]copyInfo,
 	// Deal with the single file case
 	copyInfo, err := copyInfoForFile(o.source, origPath)
 	switch {
+	case imageSource == nil && errors.Is(err, os.ErrNotExist):
+		return nil, errors.Wrapf(err, "file not found in build context or excluded by .dockerignore")
 	case err != nil:
 		return nil, err
 	case copyInfo.hash != "":
@@ -315,6 +317,10 @@ func (o *copier) copyWithWildcards(origPath string) ([]copyInfo, error) {
 func copyInfoForFile(source builder.Source, path string) (copyInfo, error) {
 	fi, err := remotecontext.StatAt(source, path)
 	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			// return the relative path in the error, which is more user-friendly than the full path to the tmp-dir
+			return copyInfo{}, errors.WithStack(&os.PathError{Op: "stat", Path: path, Err: os.ErrNotExist})
+		}
 		return copyInfo{}, err
 	}
 

--- a/integration-cli/docker_cli_build_test.go
+++ b/integration-cli/docker_cli_build_test.go
@@ -2189,18 +2189,13 @@ func (s *DockerSuite) TestBuildEntrypointRunCleanup(c *testing.T) {
 
 func (s *DockerSuite) TestBuildAddFileNotFound(c *testing.T) {
 	name := "testbuildaddnotfound"
-	expected := "foo: no such file or directory"
-
-	if testEnv.OSType == "windows" {
-		expected = "foo: The system cannot find the file specified"
-	}
 
 	buildImage(name, build.WithBuildContext(c,
 		build.WithFile("Dockerfile", `FROM `+minimalBaseImage()+`
         ADD foo /usr/local/bar`),
 		build.WithFile("bar", "hello"))).Assert(c, icmd.Expected{
 		ExitCode: 1,
-		Err:      expected,
+		Err:      "stat foo: file does not exist",
 	})
 }
 

